packages/cli/src/cmds/beacon/initBeaconState.ts

```diff
@@ -23,7 +19,9 @@ import {IBeaconArgs} from "./options.js";
 
 export function getCheckpointFromState(state: BeaconStateAllForks): Checkpoint {
   return {
-    epoch: computeEpochAtSlot(state.latestBlockHeader.slot),
+    // the correct checkpoint is based on state's slot, its latestBlockHeader's slot's epoch can be
+    // behind the state
+    epoch: Math.ceil(state.slot / SLOTS_PER_EPOCH),
```

Comment:
using `state.slot` makes sense, thru I think `computeEpochAtSlot(state.slot)` is enough

if `SLOTS_PER_EPOCH` = 32 and state.slot = 40 then this function return an epoch of 2 does not sounds right, otherwise please drop some comments to avoid the confusing

also note that there is another `getCheckpointFromState` function defined somewhere else which expect the real checkpoint state (state.slot % 32 = 0)

Reply:
my reasoning: if state slot is >  some epoch boundary (ideally it should be on the boundary), and it its a checkpoint state, it should be for next epoch+1 only right? could be possible that the latestBlockHeader belongs from > epoch*32 slot, and hence the checkpoint root might belong to epoch+1
Also this: https://github.com/ChainSafe/lodestar/blob/unstable/packages/beacon-node/src/chain/initState.ts#L226